File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        min_length = float('inf')
        current_sum = 0

        for right in range(len(nums)):
            current_sum += nums[right]

            # Shrink the window from the left as long as the condition is met
            while current_sum >= target:
                min_length = min(min_length, right - left + 1)
                current_sum -= nums[left]
                left += 1

        return min_length if min_length != float('inf') else 0

File: Data Structures & Algorithms/remove-element/submission-0.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var i = 0
        var n = nums.size
        while (i < n) {
            if (nums[i] == `val`) {
                n--
                nums[i] = nums[n]
            } else {
                i++
            }
        }

        return n
    }
}
